tests/components/azure_event_hub/test_azure_event_hub.py

```diff
@@ -0,0 +1,198 @@
+"""The tests for the Apache Kafka component."""
+from collections import namedtuple
+
+import pytest
+
+import homeassistant.components.azure_event_hub as azure_event_hub
+from homeassistant.core import split_entity_id
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+from homeassistant.setup import async_setup_component
+
+from tests.async_mock import AsyncMock, MagicMock, Mock, patch
+
+AZURE_EVENT_HUB_PATH = "homeassistant.components.azure_event_hub"
+
+
+@pytest.fixture(autouse=True, name="mock_client")
+def mock_client_fixture():
+    """Mock the azure event hub producer client."""
+    with patch(f"{AZURE_EVENT_HUB_PATH}.EventHubProducerClient") as client:
+        yield client
+
+
+@pytest.fixture(autouse=True, name="mock_event_data")
+def mock_event_data_fixture():
+    """Mock the azure event data component."""
+    with patch(f"{AZURE_EVENT_HUB_PATH}.EventData") as event_data:
+        yield event_data
+
+
+@pytest.fixture(autouse=True)
+def mock_bus_and_json(hass, monkeypatch):
+    """Mock the event bus listener and os component."""
+    hass.async_create_task = AsyncMock()
+    monkeypatch.setattr(
+        f"{AZURE_EVENT_HUB_PATH}.json.dumps", Mock(return_value=MagicMock())
+    )
+
+
+async def test_minimal_config(hass, mock_client):
+    """Test the minimal config and defaults of component."""
+    config = {azure_event_hub.DOMAIN: {"event_hub_connection_string": "connection"}}
+    assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
+
+
+async def test_full_config(hass, mock_client):
+    """Test the full config of component."""
+    config = {
+        azure_event_hub.DOMAIN: {
+            "event_hub_connection_string": "connection",
+            "send_interval": 10,
+            "max_delay": 10,
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
+    assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
+
+
+FilterTest = namedtuple("FilterTest", "id should_pass")
+
+
+def make_event(entity_id):
+    """Make a mock event for test."""
+    domain = split_entity_id(entity_id)[0]
+    state = MagicMock(
+        state="not blank",
+        domain=domain,
+        entity_id=entity_id,
+        object_id="entity",
+        attributes={},
+    )
+    return MagicMock(data={"new_state": state}, time_fired=12345)
+
+
+async def _setup(hass, filter_config):
+    """Shared set up for filtering tests."""
```

Comment:
Same comment as `apache_kafka`

Reply:
Ok wow that was way more difficult then kafka. This component doesn't actually listen for state changed events, instead it appears to listen for all events on the bus, stick them all into a queue, then process that queue on an interval. 

I got it working though, by mocking `async_call_later` I can get ahold of the method it uses for processing the queue. Then I can add events and force it to process the queue immediately so I can assert immediately after. Seems to be working, let me know if you see any concerns.

I also had to mock methods individually for this one as well, when I tried to patch `EventHubProducerClient` directly I got `TypeError: object MagicMock can't be used in 'await' expression` any time one of the methods was called. Patching the methods works fine though.

Reply:
Ideally in  `async_call_later(self.hass, 10, self.async_send)` we would make 10 a constant and then patch it to be 0, fire the event, and async_block_till_done.  That may be really hard to make work though.

Reply:
That could work for start-up but there's also the loop to worry about. Every time it finishes processing a batch it does this:
```python
if not self.shutdown:
    self._next_send_remover = async_call_later(
        self.hass, self._send_interval, self.async_send
    )
```
I don't know what I could do about that other then capturing the processor method. If I try and set this time to 0 then it'll just be continuously trying to process batches. And if I set it to anything other then 0 then I have a sleep interval to worry about in the test.

Reply:
I don't see a better way to do it.  Let's leave it as-is, and adjust if another reviewer gives feedback on a possible solution.